stan/math/prim/fun/lb_constrain.hpp

```diff
@@ -44,29 +43,27 @@ inline return_type_t<T, L> lb_constrain(const T& x, const L& lb) {
  * reference with the log absolute Jacobian determinant of the
  * transform.
  *
- * If the lower bound is negative infinity, this function
- * reduces to <code>identity_constraint(x, lp)</code>.
- *
- * @tparam T type of scalar
+ * @tparam T Type of Matrix
  * @tparam L type of lower bound
  * @tparam S type of log probability
- * @param[in] x unconstrained scalar input
+ * @param[in] x unconstrained Matrix input
  * @param[in] lb lower bound on output
  * @param[in,out] lp reference to log probability to increment
  * @return lower-bound constrained value corresponding to inputs
  */
-template <typename T, typename L, typename S>
-inline return_type_t<T, L> lb_constrain(const T& x, const L& lb, S& lp) {
-  using std::exp;
-  if (lb == NEGATIVE_INFTY) {
-    return identity_constrain(x, lp);
-  }
-  lp += x;
-  return exp(x) + lb;
+template <typename T, typename L>
+inline auto lb_constrain(const T& x, const L& lb, return_type_t<T, L>& lp) {
+  const auto& x_ref = to_ref(x);
+  const auto& lb_ref = to_ref(lb);
+  check_finite("lb_constrain", "lb", value_of(lb_ref));
+  lp += sum(x_ref);
+
+  return make_holder([](const auto& x_ref,
+                        const auto& lb_ref) { return add(exp(x_ref), lb_ref); },
+                     std::move(x_ref), std::move(lb_ref));
```

Comment:
I'd get rid of all these std moves. They are also moving from something that is supposed to be a const ref